### PR TITLE
README update in master for 202009.00 release

### PR DESCRIPTION
**Pull request by aggarw13**
Port `README.md` from `release-candidate` to `master` branch for **202009.00** release 

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,22 +118,23 @@ It is required to setup an AWS account and access the AWS IoT Console for runnin
...
-- You can pass the following configuration settings as command line options in order to run the mutual auth demos: 
+### Configuring the MQTT mutual auth demo
+
+- You can pass the following configuration settings as command line options in order to run the mutual auth demo:
 ```bash
 cmake .. -DAWS_IOT_ENDPOINT="aws-iot-endpoint" -DROOT_CA_CERT_PATH="root-ca-path" -DCLIENT_CERT_PATH="certificate-path" -DCLIENT_PRIVATE_KEY_PATH="private-key-path" 
 ```
 
-- In order to set these configurations manually, edit `demo_config.h` in `demos/mqtt/mqtt_demo_mutual_auth/` and `demos/http/http_demo_mutual_auth/` to `#define` the following:
+- In order to set these configurations manually, edit `demo_config.h` in `demos/mqtt/mqtt_demo_mutual_auth/` to `#define` the following:
```
Comment (leegeth) [inline README.md:138]: http demo details should be kept.

File: README.md [hunk 2]
```diff
@@ -41,22 +118,23 @@ It is required to setup an AWS account and access the AWS IoT Console for runnin
...
 
 
 *Note: If using the Provisioning library, a fleet provisioning template, a provisioning claim, IoT policies and IAM policies need to be setup for the AWS account. Complete the steps to setup your device and AWS IoT account outlined [here](https://docs.aws.amazon.com/iot/latest/developerguide/provision-wo-cert.html#use-claim).*
 
 
 
-### Configuring the mutual auth demos
 
-- You can pass the following configuration settings as command line options in order to run the mutual auth demos: 
+### Configuring the MQTT mutual auth demo
```
Comment (leegeth) [inline README.md:130]:
```suggestion
### Configuring the MQTT mutual auth demos
```

File: README.md [hunk 3]
```diff
@@ -41,22 +118,23 @@ It is required to setup an AWS account and access the AWS IoT Console for runnin
...
+
+- You can pass the following configuration settings as command line options in order to run the mutual auth demo:
```
Comment (leegeth) [inline README.md:132]:
```suggestion
- You can pass the following configuration settings as command line options in order to run the mutual auth demos:
```

File: README.md [hunk 4]
```diff
@@ -143,33 +221,6 @@ docker run -it -p 8883:8883 -v $(pwd):/mosquitto/config/ --name mosquitto-basic-
 
 6. Set `ROOT_CA_CERT_PATH` to the absolute path of the CA certificate created in step 3. for the local Mosquitto server.
 
-
```
Comment (leegeth) [inline README.md:146]: Http demo configuration details should be kept.

File: README.md [hunk 5]
```diff
@@ -1,12 +1,55 @@
...
 
+### AWS IoT Device Shadow
+[AWS IoT Device Shadow](https://github.com/aws/device-shadow-for-aws-iot-embedded-sdk) enables you to store and retrieve the current state (the “shadow”) of every registered device. The device’s shadow is a persistent, virtual representation of your device that you can interact with from AWS IoT Core even if the device is offline. The device state captured as its “shadow” is itself a [JSON](https://www.json.org/) document. The device can send commands over MQTT or HTTP to update its latest state. Each device’s shadow is uniquely identified by the name of the corresponding “thing”, a representation of a specific device or logical entity on the AWS Cloud. See [Managing Devices with AWS IoT](https://docs.aws.amazon.com/iot/latest/developerguide/iot-thing-management.html) for more information. More details about shadows can be found in [AWS IoT documentation](https://docs.aws.amazon.com/iot/latest/developerguide/iot-device-shadows.html). 
+
+### JSON Parser
+The coreJSON library is a JSON parser that strictly enforces the [ECMA-404 JSON standard](https://www.json.org/json-en.html).
+
```
Comment (leegeth) [inline README.md:27]: I think it is better to mention about HTTP as well here.

File: README.md (hunk 2 above)
Comment (aggarw13) [reply README.md:130]: There is only a single mutual auth demo, isn't it?

File: README.md [hunk 6]
```diff
@@ -19,12 +19,16 @@ The C-SDK simplifies access to the Pub/Sub functionality of the AWS IoT broker v
...
 
+### HTTP
+The C-SDK provides the ability to establish an HTTP connection with a server over a customer-implemented transport layer, which can either be a secure channel like a TLS session (mutually authenticated or server-only authentication) or a non-secure channel like a plaintext TCP connection. The HTTP connection can be used to make REST API requests that include "GET" (include range requests), "PUT", "POST" and "HEAD" requests. The SDK provides a mechanism to register customer-defined callback for receiving parsed header fields in an HTTP response. The [coreHTTP](https://github.com/FreeRTOS/coreHTTP) library has been refactored for memory optimization, and is a client implementation of a subset of the [HTTP/1.1](https://tools.ietf.org/html/rfc2616) standard.
```
Comment (sarenameas) [inline README.md:23]: The SDK provides a mechanism to register **a** customer-defined callback for receiving parsed header fields in an HTTP response.

File: README.md (hunk 5 above)
Comment (aggarw13) [reply README.md:30]: Added a text for the HTTP library

File: README.md (hunk 2 above)
Comment (leegeth) [reply README.md:129]: There is a mutual auth demo for http as well.

File: README.md (hunk 1 above)
Comment (leegeth) [reply README.md:137]: This is not addressed yet.

**Review by sarenameas — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -1,12 +1,59 @@
...
+# AWS IoT Device SDK for Embedded C
 
 
-## Branches
+**[API documentation of 202009.00 release](https://docs.aws.amazon.com/freertos/latest/lib-ref/embedded-csdk/202009.00/lib-ref/index.html)**    
+
+We have released version 202009.00 of the AWS IoT Device SDK for Embedded-C (C-SDK). This release includes refactored MQTT, JSON Parser, and AWS IoT Device Shadow libraries for optimized memory usage and modularity, and includes dependent libraries via GitHub submoduling. These libraries have gone through code quality checks including verification that no function has a [GNU Complexity](https://www.gnu.org/software/complexity/manual/complexity.html) score over 8, and checks against deviations from mandatory rules in the [MISRA coding standard](https://www.misra.org.uk/MISRAHome/MISRAC2012/tabid/196/Default.aspx). Deviations from the MISRA C:2012 guidelines are documented under [MISRA Deviations](MISRA.md). This library has also undergone both static code analysis from [Coverity static analysis](https://scan.coverity.com/), and validation of memory safety and proof of functional correctness through the [CBMC automated reasoning tool](https://www.cprover.org/cbmc/).
```
Comment (dan4thewin) [inline README.md:7]: "dependent"?  Are they?

"This library has also undergone" -> "These...

File: README.md [hunk 8]
```diff
@@ -1,12 +1,59 @@
...
+
+If you are upgrading from v3.x of the C-SDK to the 202009.00 release, please follow the [migration guide](https://docs.aws.amazon.com/freertos/latest/lib-ref/embedded-csdk/202009.00/lib-ref/docs/doxygen/output/html/migration_guide.html). If you are using the C-SDK v4_beta_deprecated branch, note that we will continue to maintain this branch for critical bug fixes and security patches but will not add new features to it. See the C-SDK v4_beta_deprecated branch [README](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta_deprecated/README.md) for additional details.
```
Comment (dan4thewin) [inline README.md:9]: Is there anything on the deprecated README that is helpful?  What might we say here that would be helpful?

File: README.md [hunk 9]
```diff
@@ -1,12 +1,59 @@
...
+
+If you are upgrading from v3.x of the C-SDK to the 202009.00 release, please follow the [migration guide](https://docs.aws.amazon.com/freertos/latest/lib-ref/embedded-csdk/202009.00/lib-ref/docs/doxygen/output/html/migration_guide.html). If you are using the C-SDK v4_beta_deprecated branch, note that we will continue to maintain this branch for critical bug fixes and security patches but will not add new features to it. See the C-SDK v4_beta_deprecated branch [README](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta_deprecated/README.md) for additional details. 
+
+## Overview
+
+The AWS IoT Device SDK for Embedded C (C-SDK) is a collection of C source files under the [MIT open source license](LICENSE) that can be used in embedded applications to securely connect IoT devices to [AWS IoT Core](http://docs.aws.amazon.com/iot/latest/developerguide/what-is-aws-iot.html). It includes a MQTT, JSON Parser, and AWS IoT Device Shadow library. It is distributed in source form and intended to be built into customer firmware along with application code, other libraries and RTOS.
```
Comment (dan4thewin) [inline README.md:13]: "It includes" - we need to emphasize that things like MQTT and JSON are optional...
"a MQTT" -> "an MQTT client"

**Review by dan4thewin — COMMENTED** [review]
Sorry, I didn't finish.  Submitting what I have so far.

File: README.md (hunk 2 above)
Comment (aggarw13) [reply README.md:129]: But this mentions `**MQTT** mutual auth"

Comment (leegeth) [reply README.md:129]: Yea. Sorry, my bad. I didn't notice MQTT in there. I will update the suggestion

File: README.md (hunk 8 above)
Comment (aggarw13) [reply README.md:9]: There is nothing noteworthy about the `v4_beta_deprecated` branch besides what has already been mentioned about the support of ciritical bug fixes, and no new features.

File: README.md (hunk 7 above)
Comment (aggarw13) [reply README.md:7]: >"dependent"? Are they?

I have separated the _git submoduling_ part into a separate and non-confusing statement. 

> "This library has also undergone" -> "These...

Fixed

File: README.md (hunk 9 above)
Comment (aggarw13) [reply README.md:13]: > "It includes" - we need to emphasize that things like MQTT and JSON are optional...

I have updated to mention "It _contains_ and MQTT...". What unintended interpretation of the sentence are you concerned about?

File: README.md [hunk 10]
```diff
@@ -16,56 +64,85 @@ The [v4_beta_deprecated](https://github.com/aws/aws-iot-device-sdk-embedded-C/tr
...
-### Configuring the mutual auth demos
-
-- You can pass the following configuration settings as command line options in order to run the mutual auth demos: 
+- You can pass the following configuration settings as command line options in order to run the mutual auth demo:
 ```bash
 cmake .. -DAWS_IOT_ENDPOINT="aws-iot-endpoint" -DROOT_CA_CERT_PATH="root-ca-path" -DCLIENT_CERT_PATH="certificate-path" -DCLIENT_PRIVATE_KEY_PATH="private-key-path" 
 ```
 
-- In order to set these configurations manually, edit `demo_config.h` in `demos/mqtt/mqtt_demo_mutual_auth/` and `demos/http/http_demo_mutual_auth/` to `#define` the following:
+- In order to set these configurations manually, edit `demo_config.h` in `demos/mqtt/mqtt_demo_mutual_auth/` to `#define` the following:
```
Comment (aggarw13) [inline README.md:137]:
```suggestion
- In order to set these configurations manually, edit `demo_config.h` in `demos/mqtt/mqtt_demo_mutual_auth/` and `demos/http/http_demo_mutual_auth/` to `#define` the following:
```

File: README.md (hunk 1 above)
Comment (aggarw13) [reply README.md:138]: Fixed now!

**Review by leegeth — DISMISSED** [review]

**Review by abhidixi11 — DISMISSED** [review]

File: README.md [hunk 11]
```diff
@@ -1,12 +1,60 @@
...
+
+### AWS IoT Device Shadow
+[AWS IoT Device Shadow](https://github.com/aws/device-shadow-for-aws-iot-embedded-sdk) enables you to store and retrieve the current state (the “shadow”) of every registered device. The device’s shadow is a persistent, virtual representation of your device that you can interact with from AWS IoT Core even if the device is offline. The device state captured as its “shadow” is itself a [JSON](https://www.json.org/) document. The device can send commands over MQTT or HTTP to update its latest state. Each device’s shadow is uniquely identified by the name of the corresponding “thing”, a representation of a specific device or logical entity on the AWS Cloud. See [Managing Devices with AWS IoT](https://docs.aws.amazon.com/iot/latest/developerguide/iot-thing-management.html) for more information. More details about shadows can be found in [AWS IoT documentation](https://docs.aws.amazon.com/iot/latest/developerguide/iot-device-shadows.html). 
+
+### JSON Parser
+The coreJSON library is a JSON parser that strictly enforces the [ECMA-404 JSON standard](https://www.json.org/json-en.html).
```
Comment (dan4thewin) [inline README.md:30]: How terse!

File: README.md [hunk 12]
```diff
@@ -1,12 +1,60 @@
...
+
+
+## Metrics
+
+Within the MQTT Demo, users have the ability to report operating System, hardware platform and MQTT Client information to AWS IoT by sending a specially formatted string in the username field of the MQTT CONNECT packet.
```
Comment (dan4thewin) [inline README.md:35]: This isn't a metric that the customer can consume, is it?

File: README.md [hunk 13]
```diff
@@ -1,12 +1,60 @@
...
+* **MQTT_Library_name** is the MQTT Client library being used.
+* **MQTT_Library_version** is the version of the MQTT Client library being used.
+
+
+
+## Branches
 
 ### master branch
-The [master](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/master) branch hosts the continuous development of the AWS IoT Embedded C SDK libraries. Please be aware that the libraries in master branch may have bugs and performance issues. Consider using the released versions of the AWS IoT Embedded C SDK for production ready software.
+The [master](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/master) branch hosts the continuous development of the AWS IoT Embedded C SDK libraries. Please be aware that the development at the tip of the master branch is continuously in progress and may have bugs. Consider using the tagged releases of the AWS IoT Embedded C SDK for production ready software.
```
Comment (dan4thewin) [inline README.md:57]: "AWS IoT Embedded C SDK" - this is not how we refer to it in the title.  We should either be consistent, or introduce a shortened form for convenience: C SDK or C-SDK.  (This goes for the rest of the document.)

File: README.md [hunk 14]
```diff
@@ -16,42 +64,72 @@ The [v4_beta_deprecated](https://github.com/aws/aws-iot-device-sdk-embedded-C/tr
...
+
+* Y represents the year.
+* M represents the month.
+* N represents the release order within the designated month (00 being the first release).
+* A "major" denotation indicates the addition of new features or significant updates to multiple libraries. 
+
+For example, a second release in June 2021 would be 202106.01. Although the SDK releases have moved to date-based versioning, each library within the SDK will still retain semantic versioning. In semantic versioning, the version number itself (X.Y.Z) indicates whether the release is a major, minor, or point release. You can use the semantic version of a library to assess the scope and impact of a new release on your application. 
+
+## Cloning
+This repo uses [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) to bring in dependent components.
```
Comment (dan4thewin) [inline README.md:79]: dependent?  Who depends upon whom?

File: README.md [hunk 15]
```diff
@@ -16,42 +64,72 @@ The [v4_beta_deprecated](https://github.com/aws/aws-iot-device-sdk-embedded-C/tr
...
+
+Note: If you download the ZIP file provided by GitHub UI, you will not get the contents of the submodules (The ZIP file is also not a valid git repository).  If you download from the [202009.00 Release Page](https://github.com/aws/aws-iot-device-sdk-embedded-C/releases/tag/202009.00) page, you will get the entire repository (including the submodules) in the ZIP file, aws-iot-device-sdk-embedded-c-202009.00.zip.  
+
+
+To clone using HTTPS:
+```
+git clone https://github.com/aws/aws-iot-device-sdk-embedded-C.git --recurse-submodules
```
Comment (dan4thewin) [inline README.md:86]: The git CLI usage shows the flag coming before the repository.

File: README.md [hunk 16]
```diff
@@ -16,42 +64,72 @@ The [v4_beta_deprecated](https://github.com/aws/aws-iot-device-sdk-embedded-C/tr
...
+If you have downloaded the repo without using the `--recurse-submodules` argument, you need to run:
+```
+git submodule update --init --recursive
+```
+
+
 ## Building and Running Demos
 
   
+The libraries in this SDK are not dependent on any operating system. However, the demos for the libraries in this SDK are built and tested on a Linux platform. This SDK builds with [CMake](https://cmake.org/), a cross-platform build tool.
```
Comment (dan4thewin) [inline README.md:102]: cmake is not a requirement to build the SDK itself, is it?

File: README.md [hunk 17]
```diff
@@ -16,42 +64,72 @@ The [v4_beta_deprecated](https://github.com/aws/aws-iot-device-sdk-embedded-C/tr
...
 
-The libraries in this SDK are not dependent on any operating systems. However, the demos for the libraries in this SDK are built and tested on a Linux platform. This SDK builds with [CMake](https://cmake.org/), a cross-platform build tool.
 
+### Prerequisites
 
+- CMake 3.13.0 or later
```
Comment (dan4thewin) [inline README.md:107]: We should be clear about the exact role of cmake.  We should not intimate that a customer needs cmake in order to utilize the SDK.

File: README.md [hunk 18]
```diff
@@ -16,42 +64,72 @@ The [v4_beta_deprecated](https://github.com/aws/aws-iot-device-sdk-embedded-C/tr
...
 
+### Prerequisites
 
+- CMake 3.13.0 or later 
 
-### Prerequisites
+- C90 compiler
 
-- CMake 3.13.0 or later and a C90 compiler.
+- Although not a part of the C90 standard, `stdint.h` is required for fixed-width integer types (e.g int32_t).
```
Comment (dan4thewin) [inline README.md:111]: Can we point to an alternate stdint.h for non-conforming systems, or at least say which types are needed?

File: README.md [hunk 19]
```diff
@@ -16,42 +64,72 @@ The [v4_beta_deprecated](https://github.com/aws/aws-iot-device-sdk-embedded-C/tr
...
     - Linux system with POSIX sockets and timer APIs. (CI tests on Ubuntu 18.04).
         - On Linux systems, installation of OpenSSL development libraries and header files, *version 1.1.0 or later*, are required. The OpenSSL development libraries are usually called something like `libssl-dev` or `openssl-devel` when installed through a package manager.
-        - Although not a part of the C90 standard, `stdint.h` is required for fixed-width integer types (e.g int32_t).
+        
 
 
 
 ### AWS IoT Account Setup
 
 It is required to setup an AWS account and access the AWS IoT Console for running demos and tests. Follow the links to:
```
Comment (dan4thewin) [inline README.md:122]: This is an awkward sentence.  How about "The demos and tests require an AWS account."

File: README.md (hunk 17 above)
Comment (aggarw13) [reply README.md:107]: The Prerequisites section (in which this point exists) is under the "Building and Running Demos" parent section which conveys that CMake is required for building the demos.

File: README.md (hunk 18 above)
Comment (aggarw13) [reply README.md:111]: At this point, I will choose the latter option. If we provide an alternative header file, we should test with it also.

File: README.md (hunk 14 above)
Comment (aggarw13) [reply README.md:79]: Fixed the statement to remove the "dependent" word

File: README.md (hunk 19 above)
Comment (aggarw13) [reply README.md:122]: The focus of the statement is on the reader for _needing_ to create an AWS account. I have updated the statement to use first-person voice as "You need to setup an AWS..."

File: README.md (hunk 13 above)
Comment (aggarw13) [reply README.md:57]: Updated to use C-SDK at these places

File: README.md [hunk 20]
```diff
@@ -1,12 +1,60 @@
...
+
+### HTTP
+The C-SDK provides the ability to establish an HTTP connection with a server over a customer-implemented transport layer, which can either be a secure channel like a TLS session (mutually authenticated or server-only authentication) or a non-secure channel like a plaintext TCP connection. The HTTP connection can be used to make REST API requests that include "GET" (include range requests), "PUT", "POST" and "HEAD" requests. The SDK provides a mechanism to register a customer-defined callback for receiving parsed header fields in an HTTP response. The [coreHTTP](https://github.com/FreeRTOS/coreHTTP) library has been refactored for memory optimization, and is a client implementation of a subset of the [HTTP/1.1](https://tools.ietf.org/html/rfc2616) standard.
```
[the remark this answers is not in the document]
Comment (aggarw13) [reply README.md:24]: Removed use of `REST`

File: README.md [hunk 21]
```diff
@@ -1,12 +1,60 @@
...
+
+## Overview
+
+The AWS IoT Device SDK for Embedded C (C-SDK) is a collection of C source files under the [MIT open source license](LICENSE) that can be used in embedded applications to securely connect IoT devices to [AWS IoT Core](http://docs.aws.amazon.com/iot/latest/developerguide/what-is-aws-iot.html). It contains an MQTT client, HTTP client, JSON Parser, and AWS IoT Device Shadow client libraries. It is distributed in source form and intended to be built into customer firmware along with application code, other libraries and RTOS. 
+
+## Features
+
+The C-SDK simplifies access to the Pub/Sub functionality of the AWS IoT broker via MQTT and provides APIs to interact with Device Shadow. The SDK has been tested to work with the AWS IoT Core and an open source MQTT broker to ensure best interoperability of a device with the AWS IoT platform. The C-SDK contains the following libraries:
```
[the remark this answers is not in the document]
Comment (aggarw13) [reply README.md:18]: Good suggestion, have made the change

File: README.md (hunk 18 above)
Comment (aggarw13) [reply README.md]: Have provided the list of types that we use from `stdint.h`

File: README.md [hunk 22]
```diff
@@ -1,12 +1,60 @@
...
... C-SDK provides the ability to establish an MQTT connection with a broker over a customer-implemented transport layer, which can either be a secure channel like a TLS session (mutually authenticated or server-only authentication) or a non-secure channel like a plaintext TCP connection. This MQTT connection can be used for performing publish operations to MQTT topics and subscribing to MQTT topics. The SDK provides a mechanism to register customer-defined callbacks for receiving incoming PUBLISH, acknowledgement and keep-alive response events from the broker.  The [coreMQTT](https://github.com/FreeRTOS/coreMQTT) library has been refactored for memory optimization and is fully compliant with the [MQTT 3.1.1](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/mqtt-v3.1.1.html) standard.  It has no dependencies on any additional libraries other than the standard C library, a customer-implemented network transport interface, and *optionally* a customer-implemented platform time function.  The refactored design embraces different use-cases, ranging from resource-constrained platforms using only QoS 0 MQTT PUBLISH messages to resource-rich platforms using QoS 2 MQTT PUBLISH over TLS connections.
```
[the remark this answers is not in the document]
Comment (aggarw13) [reply README.md:21]: Added a sentence under the "Features
 section above to get this message across

File: README.md (hunk 12 above)
Comment (sarenameas) [reply README.md:37]: What do you mean by "consume"? It is in the demo only and not part of the MQTT library. We found that most customers have no problem adding metrics to their MQTT message to AWS IoT themselves, if they knew how. We let customers know that these metrics help AWS IoT improve security and provide better technical support (this statement is taken from PM justification doc). 


I think we Should we add the text: "These metrics help AWS IoT improve security and provide better technical support."

Comment (dan4thewin) [reply README.md:37]: I see a title like "Metric" and I think I will be able to see data about the workings of my system.  In this case the data is for AWS, and not for the customer.  I think we should call it something other than Metric.

File: README.md [hunk 23]
```diff
@@ -1,57 +1,137 @@
...
+
+The AWS IoT Device SDK for Embedded C (C-SDK) is a collection of C source files under the [MIT open source license](LICENSE) that can be used in embedded applications to securely connect IoT devices to [AWS IoT Core](http://docs.aws.amazon.com/iot/latest/developerguide/what-is-aws-iot.html). It contains an MQTT client, HTTP client, JSON Parser, and AWS IoT Device Shadow client libraries. It is distributed in source form, and is intended to be built into customer firmware along with application code, other libraries and RTOS (Real-Time Operating System). 
+
+## Features
+
+The C-SDK simplifies access AWS IoT Services. The SDK has been tested to work with AWS IoT Core and an open source MQTT broker to ensure high quality interoperability of a device with the AWS IoT platform. 
+The AWS IoT Device Shadow library is flexible to work any MQTT library implementation. The MQTT library in the C-SDK is offered as one choice without being tightly coupled with the rest of the SDK.
+The C-SDK contains the following libraries:
+
+### MQTT
```
[the remark this answers is not in the document]
Comment (aggarw13) [reply README.md:22]: I suppose we are being generic for the libraries offered in the SDK for readability

File: README.md [hunk 24]
```diff
@@ -1,57 +1,137 @@
...
+
+## Overview
+
+The AWS IoT Device SDK for Embedded C (C-SDK) is a collection of C source files under the [MIT open source license](LICENSE) that can be used in embedded applications to securely connect IoT devices to [AWS IoT Core](http://docs.aws.amazon.com/iot/latest/developerguide/what-is-aws-iot.html). It contains an MQTT client, HTTP client, JSON Parser, and AWS IoT Device Shadow client libraries. It is distributed in source form, and is intended to be built into customer firmware along with application code, other libraries and RTOS (Real-Time Operating System). 
+
+## Features
+
+The C-SDK simplifies access AWS IoT Services. The SDK has been tested to work with AWS IoT Core and an open source MQTT broker to ensure high quality interoperability of a device with the AWS IoT platform. 
+The AWS IoT Device Shadow library is flexible to work any MQTT library implementation. The MQTT library in the C-SDK is offered as one choice without being tightly coupled with the rest of the SDK.
```
[the remark this answers is not in the document]
Comment (aggarw13) [reply README.md:19]: Done

File: README.md (hunk 12 above)
Comment (aggarw13) [reply README.md]: That's a valid concern. Have updated the section to "Collection of Metrics" and updated the text to explicitly mention that these metrics are for AWS utility, along with providing information on how these can be disabled.

File: README.md [hunk 25]
```diff
@@ -1,57 +1,137 @@
...
+If you are using the C-SDK v4_beta_deprecated branch, note that we will continue to maintain this branch for critical bug fixes and security patches but will not add new features to it. See the C-SDK v4_beta_deprecated branch [README](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta_deprecated/README.md) for additional details. 
+
+## Overview
+
+The AWS IoT Device SDK for Embedded C (C-SDK) is a collection of C source files under the [MIT open source license](LICENSE) that can be used in embedded applications to securely connect IoT devices to [AWS IoT Core](http://docs.aws.amazon.com/iot/latest/developerguide/what-is-aws-iot.html). It contains an MQTT client, HTTP client, JSON Parser, and AWS IoT Device Shadow client libraries. It is distributed in source form, and is intended to be built into customer firmware along with application code, other libraries and RTOS (Real-Time Operating System). 
+
+## Features
+
+The C-SDK simplifies access AWS IoT Services. The SDK has been tested to work with AWS IoT Core and an open source MQTT broker to ensure high quality interoperability of a device with the AWS IoT platform.
```
[the remark this answers is not in the document]
Comment (aggarw13) [reply README.md]: Updated

File: README.md (hunk 12 above)
Comment (sarenameas) [reply README.md]: Maybe "Collection of Metrics for AWS IoT"? lol

File: README.md [hunk 26]
```diff
@@ -11,12 +11,12 @@ If you are using the C-SDK v4_beta_deprecated branch, note that we will continue
...
 
 ## Features
 
-The C-SDK simplifies access AWS IoT Services. The SDK has been tested to work with AWS IoT Core and an open source MQTT broker to ensure high quality interoperability of a device with the AWS IoT platform. 
-The AWS IoT Device Shadow library is flexible to work any MQTT library implementation. The MQTT library in the C-SDK is offered as one choice without being tightly coupled with the rest of the SDK.
+The C-SDK simplifies access to AWS IoT Services. The SDK has been tested to work with AWS IoT Core and an open source MQTT broker to ensure high quality interoperability. 
+The AWS IoT Device Shadow library is flexible to work any MQTT client and JSON parser library implementations. The MQTT client and JSON parser libraries in the C-SDK are offered as one choice without being tightly coupled with the rest of the SDK.
```
Comment (dan4thewin) [inline README.md:19]: "work any" -> "work with any"
maybe drop "library implementations"
"as one choice" -> "as choices"

Comment (aggarw13) [reply README.md:19]: Fixed

**Review by dan4thewin — APPROVED** [review]

**Review by sarenameas — APPROVED** [review]

**Pull request merged**